app/models/article.rb

```diff
@@ -567,6 +568,17 @@ def set_caches
     self.path = calculated_path.downcase
   end
 
+  def normalize_title
+    return unless title
+
+    self.title = title
+      # Filter out anything that isn't a word, space, punctuation mark, or
+      # recognized emoji
+      .gsub(/[^[:word:][:space:][:punct:]\u{1f000}-\u{1ffff}]+/, " ")
+      # Coalesce runs of whitespace into a single space character
+      .gsub(/\s+/, " ")
```

Comment:
not sure if a final `strip` or `trim` makes sense (remove space on the ends), but a single space character seems generally harmless.